base/reflection.jl

```diff
@@ -955,11 +955,12 @@ function to_tuple_type(@nospecialize(t))
     t
 end
 
-function signature_type(@nospecialize(f), @nospecialize(argtypes))
+function signature_type(@nospecialize(f), @nospecialize(argtypes), check::Symbol=:type)
```

Comment:
Do you envision that the value passed as the `check` argument could be anything other than `:type` or `:callable`? As far as I can tell, as implemented currently, `:type` is used as a default value but otherwise has no particular significance since `:callable` is the only value of this argument that any code is looking for. If the only meaningful check is "is this `:callable`," perhaps it would be better suited as a `Bool` rather than a `Symbol`?

I wonder whether there is a more precise name than "callable" that could be used since anything that could meaningfully produce results from `methods` is callable. I believe "instance" is an appropriate description of what's being queried as objects of a given type are instances of that type.

Reply:
Yes, you are right - in the [interactions](https://discourse.julialang.org/t/how-to-determine-the-methods-for-a-callable-object-without-instantiating-it/102618/12?u=algunion) on the discourse topic, we also arrived at the idea that `:instance` would be more precise.

I'll address this in a future commit.